Add publishing guidelines

anvega opened the pull request:
This PR introduces a new document, guidelines.md, which outlines the publishing guidelines and standards for the CNCF Technical Advisory Group. The purpose of these guidelines is to ensure the production of high-quality, consistent, and impactful publications. Key points include content quality, structure and organization, writing style, technical accuracy, technical rigor, references and citations, review and revision, and formatting and presentation. These guidelines will help authors produce valuable resources for the community while maintaining a low-friction, enjoyable publishing process.

The guidelines emphasize the importance of considering these standards upfront to ease the burden on volunteers who handle proofing and editorial work.

brandtkeller commented:
This is a great outline for guidance and enabling the TAG to continue to iterate. 

I imagine there are a great many points that could be added - and I don't want to keep this from merge otherwise - but I do see one partially covered topic that I believe lies somewhere in the lifecycle of publications.

Do we want to include any guidance for efforts we target as high-value and as such want to keep publishing versioned updates? Do we have an obligation to do so with the items we publish or is there an expectation of best effort or otherwise deprecating after some period of time without intent to update? 

Willing to take this as a topic outside of this PR.

eddie-knight commented:
I agree with @brandtkeller that publication maintenance should at least be mentioned here. We can elaborate on it later if needed.

On another note, we should also incorporate (merge or reference) existing guidance, such as:

- governance/paper-process.md
- governance/publishing-deliverables.md

anvega commented:
When people are energized about writing, they often aspire to maintain their work long term, thinking of it as a living document. However, static versioned documents are harder to maintain than software projects. In practice, nothing gets updated until something bad enough is spotted or there’s a clear incentive for how dated something becomes. Also people’s priorities shift over time, impacting their ability to keep past work current.

For the dozen publications we have made, only the Security White Paper has seen updated versions or later editions. A few of the Supply Chain publications have had small fixes. We can mention the aspiration for ongoing updates in our guidelines but also acknowledge the reality: maintenance typically happens when there’s a compelling reason. We should encourage regular reviews and collaborative efforts while recognizing that some documents will inevitably become outdated without a strong incentive for revision.

Most importantly, I want authors to focus on writing something that meets these guidelines at the time of writing and only worry about future maintenance once they are done. Any post-completion or future maintenance plans should be included in the other two documents that focus more on the larger process/lifecycle beyond just authoring.

anvega commented:
I consolidated the three publication guidelines under a single folder and added cross-references between each. I also renamed the new doc on publishing guidelines to authoring guidelines for clarity.

mnm678 commented on publications/authoring-guidelines.md:
```diff
@@ -0,0 +1,75 @@
+
+- **Title and Abstract**: Provide a clear, concise title and abstract that summarize the main points and objectives.
+- **Sections and Headings**: Use clear and descriptive headings. Ensure that the document is divided into well-defined sections (e.g., Introduction, Background, Core Concepts, Implementation, Case Studies, Conclusion).
+- **Introduction**: Offer a compelling introduction that outlines the purpose and scope of the document.
```
really short documents (like blog posts) could maybe skip a dedicated introduction section, but I agree that it's generally needed.

anvega replied:
/s/ For really short documents, like blog posts, a dedicated introduction section might be skipped, but at least an introductory paragraph is generally recommended.

mnm678 commented on publications/authoring-guidelines.md:
```diff
@@ -0,0 +1,75 @@
+
+### 7. Review and Revision
+
+- **Peer Review**: Submit the document for peer review by knowledgeable individuals within the consortium. Incorporate feedback and revisions as needed.
```
Specifically, documents should be posted to #tag-security for feedback

anvega replied:
I added that. Thanks.

mnm678 commented on publications/publishing-protocols.md:
```diff
@@ -0,0 +1,57 @@
+- Blog posts
+- Social media outreach
+
+Publishing around KubeCon allows for broader sharing through various forums and sessions.
```
it also requires more lead time, as CNCF staff gets busy just before KubeCon

mnm678 commented on publications/publishing-protocols.md:
```diff
@@ -0,0 +1,57 @@
+
+- Review the artifact for brevity, correctness, and flow multiple times with different goals.
+- Ensure all contributors and reviewers are credited in the PDF or markdown.
+- Engage CNCF early to avoid last-minute rushes.
```
I'd add "Define the scope early in the process, and revisit it often to avoid scope creep"

mnm678 reviewed (commented):
Thanks for adding this! I left a few minor comments

anvega commented on publications/authoring-guidelines.md:
```diff
@@ -0,0 +1,75 @@
+# Authoring Guidelines and Standards
+
+The purpose of these guidelines is to ensure the production of high-quality, consistent, and impactful publications. Most pieces are authored as collaborative knowledge production efforts within working groups, under the guidance of a chair or technical leader.
+The topics tend to be sharp under the purview of a working group, for instance, “Software Supply Chain Best Practices,” “Secure Software Factory,” or “Open and Secure - A Manual for Practicing Threat Modeling to Assess and Fortify Open Source Security,” but sometimes cover broader themes like cloud native security white papers, the lexicon, and use cases and personas.
```
Replaced for specific

anvega commented on publications/authoring-guidelines.md:
```diff
@@ -0,0 +1,75 @@
+
+### 1. Content Quality
+
+- **Relevance**: Ensure all content is relevant to the topic and objectives of the publication.
```
Good shout. I incorporated that.

anvega commented on publications/paper-process.md:
```diff
@@ -0,0 +1,99 @@
+
+Adjudicators resolve community comments and suggestions. Larger discussions should be decisively resolved and documented.
+
+#### CNCF Publishing Engagement
```
This is based on existing guidelines at https://github.com/cncf/tag-security/blob/main/governance/paper-process.md. The proposed changes separate would add toil and undue delay. Chairs are expected to track ongoing work through the roadmap and should already have visibility on work in progress. Historically they've been involved through the motions ahead of final review. 

Also historically, none of the publications merged into the repo have had liaisons review the PRs prior to merging, as stipulated in the original guidelines. A few had liaisons as reviewers. Adding this additional ceremony could cause unnecessary delays, especially if we're aiming to publish by a specific date to coincide with a conference.

PushkarJ replied:
I hear you on potential delays, perhaps PR reviews were not done by TOC liaisons as described in the process in the past but I know they did review for a couple of papers I was part of through google docs and providing their approval via slack or google doc comments. 

Let me discuss this more with other chairs and get back!

anvega replied:
Part of the challenge is knowing who the liaisons are. Only the chairs are aware of this information. Even as a TL and adjudicator, I don't know who the current liaisons are. It's unreasonable to expect paper authors to know this or to pursue them for review.

anvega replied:
Also, while I understand the perceived value of executive-like reviews, this approach introduces unnecessary authority and control, leading to bureaucracy that contradicts the egalitarian ethos/principles of open source. Imposing a top-down, rigid system undermines the collaborative and fluid nature we strive for. We should avoid creating barriers that slow down innovation and publication, especially when aligning with critical timelines.

PushkarJ replied:
Chairs discussed this today and also agree with the fact that asking lead authors to connect with liaisons for review is unfair. There is a path to still request TOC liaison review that doesn't delay things as part of community feedback step/stage, when needed where chairs take up the responsibility of facilitating this review. 

I will add a follow up PR to address this in the coming days. For now this PR is a great improvement in many other aspects so we should merge this in the spirit of incremental progress. 

cc @mnm678 @eddie-knight

JonZeolla commented:
Liking the direction and overall flow

I know the compliance WG has a number of white paper ideas currently, so CC @rficcaglia , Anca, etc. for thoughts/feedback
